crates/turborepo-lockfiles/src/berry/mod.rs

```diff
@@ -461,103 +491,8 @@ impl Lockfile for BerryLockfile {
-            let locator = Locator::try_from(key.as_str()).map_err(Error::from)?;
-
-            let package = self
-                .locator_package
-                .get(&locator)
-                .cloned()
-                .ok_or_else(|| Error::MissingPackageForLocator(locator.as_owned()))?;
-
-            for (name, range) in package.dependencies.iter().flatten() {
-                let dependency = self.resolve_dependency(&locator, name, range.as_ref())?;
-                let dep_locator = self
-                    .resolutions
-                    .get(&dependency)
-                    .ok_or_else(|| Error::MissingLocator(dependency.clone().into_owned()))?;
-                resolutions.insert(dependency, dep_locator.clone());
-            }
-
-            // If the package has an associated patch we include it in the subgraph
-            if let Some(patch_locator) = self.patches.get(&locator) {
-                patches.insert(locator.as_owned(), patch_locator.clone());
-            }
-        }
-
-        for patch in patches.values() {
-            let patch_descriptors = reverse_lookup
-                .get(patch)
-                .unwrap_or_else(|| panic!("Unable to find {patch} in reverse lookup"));
-
-            // For each patch descriptor we extract the primary descriptor that each patch
-            // descriptor targets and check if that descriptor is present in the
-            // pruned map and add it if it is present
-            for patch_descriptor in patch_descriptors {
-                let version = patch_descriptor.primary_version().unwrap();
-                let primary_descriptor = Descriptor {
-                    ident: patch_descriptor.ident.clone(),
-                    range: version.into(),
-                };
-
-                if resolutions.contains_key(&primary_descriptor) {
-                    resolutions.insert((*patch_descriptor).clone(), patch.clone());
-                }
-            }
-        }
-
-        // Add any descriptors used by package extensions
-        for descriptor in &self.extensions {
-            let locator = self
-                .resolutions
-                .get(descriptor)
-                .ok_or_else(|| Error::MissingLocator(descriptor.to_owned()))?;
-            resolutions.insert(descriptor.clone(), locator.clone());
-        }
-
-        Ok(Box::new(Self {
-            data: self.data.clone(),
-            resolutions,
-            patches,
-            // We clone the following structures without any alterations and
-            // rely on resolutions being correctly pruned.
-            locator_package: self.locator_package.clone(),
-            resolver: self.resolver.clone(),
-            extensions: self.extensions.clone(),
-            overrides: self.overrides.clone(),
-            workspace_path_to_locator: self.workspace_path_to_locator.clone(),
-        }))
+        let subgraph = self.subgraph(workspace_packages, packages)?;
```

Comment:
This is an oopsie from 6 months ago where we ended up having duplicate definitions for `subgraph` one on `BerryLockfile` and one on the `Lockfile` implementation. This unifies them by forwarding the trait implementation to the type one.